server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -228,24 +229,19 @@ public ClusterState(
         this.customs = customs;
         this.wasReadFromDiff = wasReadFromDiff;
         this.routingNodes = routingNodes;
-        assert assertConsistentRoutingNodes(routingTable, nodes, routingNodes);
+        assertConsistentRoutingNodes(routingTable, nodes, routingNodes);
```

Comment:
This pattern of asserting on a boolean method that only returns `true` is pervasive - it means that when assertions are disabled we don't even invoke the method. Of course in theory this should be equivalent to checking `Assertions.ENABLED` at the start of the method, but in practice it may not be. I'd prefer we left this as written.

Reply:
I tend to agree with David here, especially if the pattern is pervasive in the code base

Reply:
That makes sense. I'll close this PR as unnecessary.